merlin/systems/dag/ops/fil.py

```diff
@@ -440,31 +402,39 @@ class XGBoost(FILModel):
     model_filename = "xgboost.json"
 
     def __init__(self, model):
-        self.model = model
-        learner = json.loads(model.save_config())["learner"]
-        self._learner = learner
-
-        objective = learner["objective"]["name"]
-        if objective == "binary:hinge":
-            raise ValueError(
-                "Objective binary:hinge is not supported."
-                "Only sigmoid and identity values of pred_transform are supported"
-                " for binary classification."
-            )
+        if model:
```

Comment:
what is the scenario where the model not get passed?

Reply:
in when we call from_config, we create a new version of the operator that is not loaded with the model. The model is read in by the config in triton. Which makes it where the only thing the operator does in the triton context is leverage the business logic scripting and sends a request to triton fil model.